classes/Product.php

```diff
@@ -422,7 +422,7 @@ class ProductCore extends ObjectModel
             'id_supplier' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId'],
             'reference' => ['type' => self::TYPE_STRING, 'validate' => 'isReference', 'size' => 64],
             'supplier_reference' => ['type' => self::TYPE_STRING, 'validate' => 'isReference', 'size' => 64],
-            'location' => ['type' => self::TYPE_STRING, 'validate' => 'isReference', 'size' => 64],
+            'location' => ['type' => self::TYPE_STRING, 'validate' => 'isString', 'size' => 255],
```

Comment:
```suggestion
            'location' => ['type' => self::TYPE_STRING, 'validate' => 'isString', 'size' => 255, 'allow_null' => true],
```

Reply:
Please update phpdoc for attribute

Reply:
removed `location` from definition. Its read-only now in product and is loaded from stockAvailable

Reply:
since the DB allows for null value it should be present in the definition indeed